### ISSUE TITLE
[external CONTACT] Issues Accessing the API

### ISSUE THREAD

**Issue by pccollins2** [ArctosDB/arctos#8136]
Hi Dusty & Michelle,

I figured I'd move the issues from our email thread to here. Following up from my recent email, here is the current issue with UAMNH. You mentioned the previous semester's group (with Ben Brunson) was under the IP of 129.114.*.*, which I have no knowledge of where that IP came from. Does the entire project need to be moved to the IP we are currently using? I've included my reply to your previous email below:

I am unsure as to whether or not the IP address within the AIME Building or UA domain has changed, but our external IP addresses from the computers we are currently working with is 130.160.124.*

Here's the error we get in Visual Studio when trying to search through the front end while running the webpage locally:

System.AggregateException: 'One or more errors occurred. (Response status code does not indicate success: 401 (Unauthorized).)'

**Comment by dustymc** [comment]
> IP of 129.114.., which I have no knowledge

That's the production UI - it's still up so I presume someone is maintaining it and wants it up?! @jcabbott72 


```
dig +short arctos.museums.ua.edu
129.114.97.203
```

>System.AggregateException: 'One or more errors occurred. (Response status code does not indicate success: 401 (Unauthorized).)'

That's not coming from the Arctos API, if any diagnostics are necessary I'll need to know what's actually happening.

@mkoo (traveling and may be slow to respond) I think this is a request for a dev key for 130.160.124.* but I'm hesitant to just issue until questions are cleared up.

**Comment by pccollins2** [comment]
After looking further into the situation, that 129.114.97.203 address is the server (TACC) hosting the web application all the way in Texas.

I'm assuming we would need a dev key for 130.160.124.*, but I am unsure as to why this hasn't been an issue for previous teams

**Comment by pccollins2** [comment]
Here's another view of the error, but on the webpage itself
![Screenshot 2024-09-26 170331](https://github.com/user-attachments/assets/86b8be65-8b3b-4cd9-9ab1-76bcc1d04253)

**Comment by pccollins2** [comment]
Here is a tracert from our 130.*.*.* address to http://arctos-test.tacc.utexas.edu/component/api/v2/catalog.cfc?method=getCatalogData

![image](https://github.com/user-attachments/assets/6aa15596-d8ad-4805-818b-ac369f90d0d2)

**Comment by dustymc** [comment]
Thanks, that helps (but text would be nice!).

I've emailed a new key and test endpoint to @jcabbott72.

**Comment by pccollins2** [comment]
Hi Dusty,

Thanks for getting back with a new key. Unfortunately, we are still getting an error with the new key/link (thankfully it's not the same one as before).


System.AggregateException: 'One or more errors occurred. (Invalid URI: Invalid port specified.)'

**Comment by dustymc** [comment]
I need to know the request to and response from Arctos in order to provide any information.

**Comment by pccollins2** [comment]
An unhandled exception occurred while processing the request.
AggregateException: One or more errors occurred. (Invalid URI: Invalid port specified.)
System.Threading.Tasks.Task.ThrowIfExceptional(bool includeTaskCanceledExceptions)

UriFormatException: Invalid URI: Invalid port specified.
System.Uri.CreateThis(string uri, bool dontEscape, UriKind uriKind, ref UriCreationOptions creationOptions)

Stack Query Cookies Headers Routing
AggregateException: One or more errors occurred. (Invalid URI: Invalid port specified.)
System.Threading.Tasks.Task.ThrowIfExceptional(bool includeTaskCanceledExceptions)
System.Threading.Tasks.Task<TResult>.GetResultCore(bool waitCompletionNotification)
System.Threading.Tasks.Task<TResult>.get_Result()
Capstone_Museum.Controllers.SearchResultsController.EthnologyResults(Search searchObj) in SearchResultsController.cs
+
			List<SearchResult> results = SearchResult.GetResults(searchObj, "ALMNH:EH").Result;
lambda_method73(Closure , object , object[] )
Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor+SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, object controller, object[] arguments)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(ref State next, ref Scope scope, ref object state, ref bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

Show raw exception details
UriFormatException: Invalid URI: Invalid port specified.
System.Uri.CreateThis(string uri, bool dontEscape, UriKind uriKind, ref UriCreationOptions creationOptions)
System.Uri..ctor(string uriString, UriKind uriKind)
System.Net.Http.HttpClient.CreateUri(string uri)
System.Net.Http.HttpClient.GetStringAsync(string requestUri)
Capstone_Museum.Models.SearchResult.GetResults(Search search, string guidPrefix) in SearchResult.cs
+
			string response = await client.GetStringAsync(url); //return object from api call

Show raw exception details

**Comment by pccollins2** [comment]
Is this what you needed, or were you referring to something else? If not, let me know where I can find it

**Comment by dustymc** [comment]
That is not what I need, Arctos did not make that. I need the actual http request and response.

**Comment by pccollins2** [comment]
This is what I got as a response in Postman. Still not sure if this is what you're looking for, so if not, please direct me on which specific parts you need and where I can find them.


	
<link type='text/css' rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css">
<script type='text/javascript' language="javascript" src='https://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js'></script>
<script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.2/jquery-ui.min.js"></script>
<link rel="stylesheet" href="https://code.jquery.com/ui/1.9.2/themes/base/jquery-ui.css" />
<script type='text/javascript' language="javascript" src='/includes/ajax.min.js?v=10.62'></script>
<link rel="stylesheet" href="/includes/style.min.css?v18" />
<!DOCTYPE html>
<html>
<head>
	<meta charset="UTF-8" />
	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
	<meta name="viewport" content="width=device-width, initial-scale=1.0" />
	<script>
		$(document).ready(function() {
			$(document).on('click','#menu_toggle',function(){
		    	$("#nav_menu_content").toggle();
		    });
			var phlp=$("#thisPagePageHelp").val();
			//console.log(phlp);
			if (phlp!=null && phlp.length>0){
				var thisPgHlp='<a target="_blank" class="external" href="' + phlp + '">Page Help</a>';
				//console.log(thisPgHlp);
				$("#pageHelpLi").html(thisPgHlp).show();
			} else {
				$("#pageHelpLi").html('').hide();
			}
			$(document).on('click','#closeannouncement',function(){			
				$.ajax({
					url: "/component/functions.cfc",
					type: "POST",
					dataType: "json",
					data: {
						method:  "dismiss_announcement",
						returnformat : "json"
					}
				});	
				$("#arctos_header_announcement").remove();
			});
			$(document).on('click','#btn_agree_terms',function(){			
				$.ajax({
					url: "/component/functions.cfc",
					type: "POST",
					dataType: "json",
					data: {
						method:  "agree_terms",
						returnformat : "json"
					},
					success: function(r) {
						location.reload();
					}
				});	
			});	
		});
	</script>
<title>Database Access</title></head>
<body>
	<div id="the_whole_header">
		<div id="nav_menu">
			<div id="arctos_logo">
				<a href="/"><img id="arctos_logo" src="/images/arctosandwords.png" alt="Arctos" border="0"></a>
			</div>
			<div id="nav_menu_hamburger">
				<span id="menu_toggle">&#9776;</span>
			</div>
			<div id="nav_menu_content">
				<div class="nav_menu_item hdr_dropdown">
					<div class="drop_menu_label">
						<div class="drop_menu_label_txt">
							<a target="_top" href="/search.cfm">Search</a> <i class="fa fa-caret-down"></i>
						</div>
						<div class="drop_list nav_menu_drop">
							<ul>
								<li><a target="_top" href="/agent.cfm">Agents</a></li>
								<li><a target="_top" href="/search.cfm">Catalog&nbsp;Records</a></li>
				                <li><a target="_top" href="/info/ctDocumentation.cfm">Code&nbsp;Tables</a></li>
								<li><a target="_top" href="/home.cfm">Collections</a></li>
				                <li><a target="_top" href="/MediaSearch.cfm">Media&nbsp;&&nbsp;Documents</a></li>
				                <li><a target="_top" href="/place.cfm">Places&nbsp;&&nbsp;Events</a></li>
								<li><a target="_top" href="/SpecimenUsage.cfm">Publications&nbsp;&&nbsp;Projects</a></li>
								<li><a target="_top" href="/random.cfm">Surprise&nbsp;Me!</a></li>
								<li><a target="_top" href="/taxonomy.cfm">Taxonomy</a></li>
								<li><a target="_top" href="/info/api.cfm">API</a></li>
							</ul>
						</div>
					</div>
				</div>
				<div class="nav_menu_item">
					<a href="/directory.cfm">Tools Directory</a>
				</div>
				<div class="nav_menu_item hdr_dropdown">
					<div class="drop_menu_label">
						<div class="drop_menu_label_txt">
							Join <i class="fa fa-caret-down"></i>
						</div>
						<div class="drop_list nav_menu_drop">
							<ul>
								<li>
									<a class="external" href="https://arctosdb.org/what-is-arctos/">What Arctos Values & Delivers</a>
								</li>
								<li>
									<a href="https://docs.google.com/forms/d/e/1FAIpQLSec0BuRKDGDjqxT1fRI31GbZc0TORLK6DoLjZReQfnl5iIyDA/viewform" target="_blank" class="external">
										Prospective Collection Request
									</a>
								</li>
								<li>
									<a target="_top" href="/info/mentor.cfm">Find a Mentor</a></li>
									<li><a class="external" href="https://handbook.arctosdb.org/how_to/new-collection.html##existing-institutions">Existing Institutions - New Portal Request</a>
								</li>
							</ul>
						</div>
					</div>
				</div>
				<div class="nav_menu_item hdr_dropdown">
					<div class="drop_menu_label">
						<div class="drop_menu_label_txt">
							Help <i class="fa fa-caret-down"></i>
						</div>
						<div class="drop_list nav_menu_drop rightdrop">
							<ul>
								<li><a href="https://arctosdb.org/acknowledgment-of-harmful-content" class="external">Acknowledgment of Harmful Content</a></li>
								<li><a target="_blank" class="external" href="http://arctosdb.org/">About</a></li>
								<li><a target="_blank" class="external" href="http://handbook.arctosdb.org/">Help</a></li>
								<li><a target="_blank" class="external" href="https://arctosdb.org/learn/webinars/">Webinars</a></li>
								<li><a target="_blank" class="external" href="https://arctosdb.org/learn/tutorial-blitz/">Tutorials</a></li>
								<li id="pageHelpLi" style="display: none;"></li>
							</ul>
						</div>
					</div>
				</div>
				
					<div class="nav_menu_item">
						
						<a href="#" onclick="openOverlay('/form/loginformguts.cfm','Log in, log out, create account, or recover password.');">Log In or Create Account</a>
					</div>
				
			</div>
		</div>
		
		
		
			<div id="arctos_header_announcement">
				<div class="arctos_announcement_text">
					Welcome to test, where everything's made up and the data don't matter.
					<span id="closeannouncement">&times;</span>
				</div>
			</div>
		
	</div>
	
	








	<div id="dv_auth_required">
		<div style="border:10px solid #F5C03D; margin: 1em;padding: 1em;">
			<p>Welcome to <a href="https://arctosdb.org/" class="external">Arctos</a>!</p>
			<p>
				By accessing this content, you are agreeing to all terms of the <a href="https://arctosdb.org/arctosdata-policy/" class="external">Arctos Community Data Policy</a> and indicating that you are aware of our <a href="https://arctosdb.org/acknowledgment-of-harmful-content/" class="external">Acknowledgment of Harmful Content</a>.
			</p>
			<p>
				Please start with the <a href="https://arctosdb.org/arctos-api-policy/" class="external">Arctos API Policy</a> for access by automation.
			</p>
			
			<div style="display:flex;flex-wrap:wrap;gap:2em;">
				<div>
					<input type="button" id="btn_agree_terms" value="I agree, continue">
				</div>
				<div>
					<input type="button" onclick="openOverlay('/form/loginformguts.cfm?action=createAccount','Create Account');" value="I agree, create account">
				</div>
				<div>
					<input type="button" onclick="openOverlay('/form/loginformguts.cfm','Log In');" value="I agree, log in">
				</div>
			</div>
		</div>
		<p>You must log in, create an account, or agree to the terms above to view this content.</p>
		<div id="arctosfooter">
	<div id="footerharmfulcontentdiv">
		<a href="https://arctosdb.org/acknowledgment-of-harmful-content" class="external">Acknowledgment of Harmful Content</a>
	</div>
	<div id="footerdonatediv">
		<a href="https://give.communityin.org/arctos-friends" class="external">Donate</a>
	</div>
	<div id="bussupt">
		<a class="external" href="https://github.com/ArctosDB/arctos/issues/new?assignees=&labels=contact&template=contact-arctos.md&title=%5BCONTACT%5D">
			Report a bug or request support
		</a>
	</div>
	<div id="altcontact">
		<a class="external" href="https://arctosdb.org/contacts/">Alternate Contacts</a>
	</div>
	<div id="cistmnt">
		<a href="https://arctosdb.org" class="external">Arctos Consortium</a>
		is a fiscally sponsored project of 
		<a href="https://www.communityin.org/" class="external">Community Initiatives</a>, a US 501(c)(3) nonprofit organization.
	</div>
	<div id="footerlogo">
		<a href="/"><img id="arctosfooterlogo" src="/images/ArctosBluegl.svg" alt="Arctos" border="0" height="50px;"></a>
	</div>
</div>
<div id="google_translate_element"></div>
<script type="text/javascript" src="//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit"></script> 
<script>
window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
ga('create', 'UA-315170-1', 'auto');
ga('send', 'pageview');
function googleTranslateElementInit(){new google.translate.TranslateElement({pageLanguage: 'en'}, 'google_translate_element');}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>

	


	
	

</body>
</html>
	</div>

**Comment by pccollins2** [comment]
{"error":"Unauthorized","Message":"Invalid API key: 18E3D79B-A36F-4665-BBB9057DF5AC4F8C from 130.160.124.1","recordsTotal":"null","recordsFiltered":"null","draw":1}

**Comment by dustymc** [comment]
That's better, I'm not sure what happened, I bounced the api, try again please.

**Comment by pccollins2** [comment]
Still getting the same thing and a 401 error

**Comment by dustymc** [comment]
What URI are you attempting to access?

**Comment by pccollins2** [comment]
https://web.corral.tacc.utexas.edu:9013 along with the test API key you provided John Abbot

**Comment by dustymc** [comment]
Thanks, try now.

**Comment by dustymc** [comment]
(and please don't post that URL in a public repo)

**Comment by pccollins2** [comment]
Nothing, still getting the same result/error.

I apologize, I thought the key was the only sensitive part

**Comment by dustymc** [comment]
How about now? (And sorry for the "wiggle the wires" experience! I fixed a bug, it works for me, I think something's cached, hopefully restarted the last bit of that now.)

**Comment by pccollins2** [comment]
Still nothing :(

**Comment by dustymc** [comment]
Blargh, how about now?

**Comment by pccollins2** [comment]
No :(

**Comment by dustymc** [comment]
You are not doing anything at all that I can see. Maybe the stuck cache is on your end?!

I dropped the IP restrictions on the key,

{the URL I send you}/component/api/v2/catalog.cfc?method=getCatalogData&guid_prefix=ALMNH:Paleo&queryformat=struct&api_key=18E3D79B-.....

returns data for me.

Do you have some time we can zoom tomorrow? I'd like to see your screen and my logs at the same time.

**Comment by pccollins2** [comment]
Yes, 5:00pm central works best for my team, but I can make 2:00pm work if needed

**Comment by dustymc** [comment]
3 Pacific works.
